pkgs/servers/grakn/grakn.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, lib, openjdk, fetchurl, graknHome ? "~/.grakn_home", fetchzip}:
+
+let 
+  graknDirLinux = "grakn-core-all-linux-2.0.0-alpha-9";
+  graknDirMac = "grakn-core-all-mac-2.0.0-alpha-9";
+  graknDirWindows = "grakn-core-all-windows-2.0.0-alpha-9";
+  graknDir = if stdenv.hostPlatform.isWindows then graknDirWindows
+             else if stdenv.isDarwin          then graknDirMac
+                                              else graknDirLinux;
+  linuxSrc = builtins.fetchTarball {
+    url = "https://github.com/graknlabs/grakn/releases/download/2.0.0-alpha-9/grakn-core-all-linux-2.0.0-alpha-9.tar.gz";
+    sha256 = "04g5n6nhffs1gc8gzmrblz6sv74sgi9c9vk6b3dsh2133zm8h8mk";
+  };
+  windowsSrc = fetchzip {
+    url = "https://github.com/graknlabs/grakn/releases/download/2.0.0-alpha-9/grakn-core-all-windows-2.0.0-alpha-9.zip";
+    sha256 = "05gzpxwyscj63lqcl2nydfzqq7911a0wai2gm6hzqjqdadl9r363";
+  };
+  macSrc = fetchzip {
+    url = "https://github.com/graknlabs/grakn/releases/download/2.0.0-alpha-9/grakn-core-all-mac-2.0.0-alpha-9.zip";
+    sha256 = "1w51anjd1gwsjqawjkd6yx4m0z5vd1cryap2i30ysm5jca8afn5j";
+  };
+  srcFolder = if stdenv.hostPlatform.isWindows then windowsSrc
+              else if stdenv.isDarwin          then macSrc
+                                               else linuxSrc ;
+in
+stdenv.mkDerivation rec {
+  pname = "grakn";
+  version = "2.0.0-alpha-9";
+
+  src = srcFolder;
+
+  buildDepends = [ openjdk ];
+
+  buildPhase = ''
+    graknFile=./${graknDir}/grakn
+    # replace the paths of the grakn boot script with the nix paths
+    sed -i "20s#^JAVA_BIN=java\$#JAVA_BIN=${openjdk}/bin/java#" ${graknDir}/grakn
+    sed -i "85s#java#${openjdk}/bin/java#" ${graknDir}/grakn
+    sed -i "54s#java#${openjdk}/bin/java#" ${graknDir}/grakn
+    sed -i "79s#java#${openjdk}/bin/java#" ${graknDir}/grakn
```

Comment:
Can we just replace all java with the nix path?

Reply:
not really, the word java is in the original script way more often, sometimes in method names, sometimes in strings or variable names;
I thought about that because it would have been easier but from my POV I had to go with this approach to not break anything

Reply:
Maybe we should convert this into a patch. sed silently fails if any of the codes changes.

Reply:
I [could use awk instead](https://stackoverflow.com/a/12144342/4440871)
or i could add a failsafe that throws an error if the file has not changed after any of the sed calls

Reply:
@SuperSandro2000 i changed the sed replacements to awk now, letting the build fail when the replacements could not be done.